tests/python/gaia-ui-tests/gaiatest/apps/phone/regions/keypad.py

```diff
@@ -74,7 +74,7 @@ class Keypad(BaseKeypad, Phone):
 
     def __init__(self, marionette):
         Phone.__init__(self, marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == self.name)
+        self.wait_for_condition(lambda m: self.apps.displayed_app.manifest_url == self.manifest_url)
```

Comment:
you can use self.wait_to_be_displayed() here. But you need a manifest_url property on the phone app.